### PR TITLE
Copy service unit configuration to Ubuntu Xenial default location

### PR DESCRIPTION
**Pull request by fuzzball5000**
Requested by @katherinealbany under #2160, copy service configuration to Ubuntu Xenial default location /etc/default and change service unit to reference the same.

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: tools/deployment/make_linux_package.sh [hunk 1]
```diff
@@ -125,7 +126,12 @@ function main() {
     cp $OSQUERY_CONFIG_SRC $INSTALL_PREFIX/$OSQUERY_ETC_DIR/osquery.conf
   fi
 
-  if [[ $DISTRO = "centos7" || $DISTRO = "rhel7" || $DISTRO = "xenial" ]]; then
+  if [[ $DISTRO = "xenial" ]]; then
+  #Change config dir path for Xenial
+  sed -i 's/sysconfig/default/g' $SYSTEMD_SERVICE_SRC
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: Why are you changing the `SRC` here?

```bash
SYSTEMD_SERVICE_SRC="$SCRIPT_DIR/osqueryd.service"
```

Would become:
```bash
SYSTEMD_SERVICE_SRC="$SCRIPT_DIR/osqueryd.service"
```

No change! I think you want `SYSTEMD_SYSCONFIG_DST`

Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: Also these two lines are missing indentation 💃

File: tools/deployment/make_linux_package.sh [hunk 2]
```diff
@@ -125,7 +126,12 @@ function main() {
...
+  fi
+
+  if [[ $DISTRO = "centos7" || $DISTRO = "rhel7" ]]; then
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: Assuming you did want to change `DST` then you can keep the include for `xenial` and make the directories. Otherwise `xenial` will have the Upstart service installed via the `else`.

File: tools/deployment/make_linux_package.sh [hunk 3]
```diff
@@ -136,8 +142,20 @@ function main() {
     cp $INITD_SRC $INSTALL_PREFIX$INITD_DST
   fi
 
-  log "creating package"
-  IFS=',' read -a deps <<< "$PACKAGE_DEPENDENCIES"
+  if [[ $DISTRO = "xenial" ]]; then
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: This repeats most of the logic above, please remove it.

File: tools/deployment/make_linux_package.sh [hunk 4]
```diff
@@ -136,8 +142,20 @@ function main() {
...
+    cp $SYSTEMD_SERVICE_SRC $INSTALL_PREFIX$SYSTEMD_SERVICE_DST
+    cp $SYSTEMD_SYSCONFIG_SRC $INSTALL_PREFIX$SYSTEMD_SYSCONFIG_DST_DEBIAN
+  else
+    mkdir -p `dirname $INSTALL_PREFIX$INITD_DST`
+    cp $INITD_SRC $INSTALL_PREFIX$INITD_DST
+  fi
+
+  log "creating package"  
+
+IFS=',' read -a deps <<< "$PACKAGE_DEPENDENCIES"
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: This needs indentation, and it seems a newline snuck in right above. :P

**Comment by fuzzball5000** [comment]
Ooops, thought it all seemed to be going too smoothly for a Monday morning after only one coffee... Shall fix!

**Comment by fuzzball5000** [comment]
That's hopefully a more sane solution...!

File: tools/deployment/make_linux_package.sh [hunk 5]
```diff
@@ -136,6 +137,15 @@ function main() {
     cp $INITD_SRC $INSTALL_PREFIX$INITD_DST
   fi
 
+  if [[ $DISTRO = "xenial" ]]; then
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: What if you add this check for `xenial` on line 131 then do:
```bash
SYSTEMD_SYSCONFIG_DST=$SYSTEMD_SYSCONFIG_DST_DEBIAN
```

Then add another check around line 134 and do:
```
sed -i 's/sysconfig/default/g' $INSTALL_PREFIX$SYSTEMD_SERVICE_DST
```

would that capture all the changes? It's 2 additional if statements, which is lame, but only 2 effective statements-- better than having to clean up the RHEL-based config.

File: tools/deployment/make_linux_package.sh [hunk 6]
```diff
@@ -51,7 +52,7 @@ DEBUG_PREFIX=$WORKING_DIR/debug
 
 function usage() {
   fatal "Usage: $0 -t deb|rpm -i REVISION -d DEPENDENCY_LIST
-
+
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: Ditch this too ;)

Comment (fuzzball5000) [inline tools/deployment/make_linux_package.sh]: I should have my return key taken away from me until I learn. Or whichever finger keeps hitting it surgically shortened...

File: tools/deployment/make_linux_package.sh (hunk 5 above)
Comment (fuzzball5000) [inline tools/deployment/make_linux_package.sh]: That would do it, the intention of my last solution was to encapsulate all of this into a single if statement, but two effective statements is more efficient.

File: tools/deployment/make_linux_package.sh [hunk 7]
```diff
@@ -51,7 +52,6 @@ DEBUG_PREFIX=$WORKING_DIR/debug
 
 function usage() {
   fatal "Usage: $0 -t deb|rpm -i REVISION -d DEPENDENCY_LIST
-
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: Why is this line deleted?

Comment (fuzzball5000) [inline tools/deployment/make_linux_package.sh]: I take a copy of the file from the project master in Sublime, edit, push up to my dev server, build test then commit. Why this process is causing lines to disappear is a mystery. Adding back in now.

Comment (fuzzball5000) [inline tools/deployment/make_linux_package.sh]: Ok, save for the additional single line of whitespace surrounding the changes, any stray additions/deletions should now be resolved.

Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: I'm sorry this is painful, haha! But the line you added back includes spaces. You'll need to add the newline without any trailing spaces for the diff/patch to ignore the change.

If you configure Sublime to show whitespace it's easier to detect these things. :) You may be able to configure Sublime to show changes within the editor view-- but I know Atom will do this by default. When you show staged changes it's helpful to pick up on random inserts/removes, which are usually caused by stray keystokes or missed key combinations when attempting to window-tab and such.

Comment (fuzzball5000) [inline tools/deployment/make_linux_package.sh]: This is as confusing as it is infuriating ;) I've never had these issues with any other files or commits before, and I've used Sublime for years. Just examined the file in both vim and Atom, there is no trailing space showing. Output from cat -A is below:

Comment (fuzzball5000) [inline tools/deployment/make_linux_package.sh]: Ignore me, it's there!

Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: After rebase squashing your commits from master here is the effective patch:
```diff
diff --git a/tools/deployment/make_linux_package.sh b/tools/deployment/make_linux_package.sh
old mode 100755
new mode 100644
index b2d8c1a..6db5765
--- a/tools/deployment/make_linux_package.sh
+++ b/tools/deployment/make_linux_package.sh
@@ -34,6 +34,7 @@ SYSTEMD_SERVICE_SRC="$SCRIPT_DIR/osqueryd.service"
 SYSTEMD_SERVICE_DST="/usr/lib/systemd/system/osqueryd.service"
 SYSTEMD_SYSCONFIG_SRC="$SCRIPT_DIR/osqueryd.sysconfig"
 SYSTEMD_SYSCONFIG_DST="/etc/sysconfig/osqueryd"
+SYSTEMD_SYSCONFIG_DST_DEBIAN="/etc/default/osqueryd"
 CTL_SRC="$SCRIPT_DIR/osqueryctl"
 PACKS_SRC="$SOURCE_DIR/packs"
 PACKS_DST="/usr/share/osquery/packs/"
@@ -51,7 +52,7 @@ DEBUG_PREFIX=$WORKING_DIR/debug
 
 function usage() {
   fatal "Usage: $0 -t deb|rpm -i REVISION -d DEPENDENCY_LIST
-
+  
   This will generate an Linux package with:
   (1) An example config /var/osquery/osquery.example.config
   (2) An init.d script /etc/init.d/osqueryd
@@ -120,11 +121,16 @@ function main() {
   cp $OSQUERY_EXAMPLE_CONFIG_SRC $INSTALL_PREFIX$OSQUERY_EXAMPLE_CONFIG_DST
   cp $PACKS_SRC/* $INSTALL_PREFIX/$PACKS_DST
 
-  if [ $OSQUERY_CONFIG_SRC != "" ] && [ -f $OSQUERY_CONFIG_SRC ]; then
+  if [[ $OSQUERY_CONFIG_SRC != "" ]] && [[ -f $OSQUERY_CONFIG_SRC ]]; then
     log "config setup"
     cp $OSQUERY_CONFIG_SRC $INSTALL_PREFIX/$OSQUERY_ETC_DIR/osquery.conf
   fi
-
+  
+  if [[ $DISTRO = "xenial" ]]; then
+    #Change config path to Ubuntu/Xenial default
+    SYSTEMD_SYSCONFIG_DST=$SYSTEMD_SYSCONFIG_DST_DEBIAN
+  fi 
+     
   if [[ $DISTRO = "centos7" || $DISTRO = "rhel7" || $DISTRO = "xenial" ]]; then
     # Install the systemd service and sysconfig
     mkdir -p `dirname $INSTALL_PREFIX$SYSTEMD_SERVICE_DST`
@@ -135,6 +141,11 @@ function main() {
     mkdir -p `dirname $INSTALL_PREFIX$INITD_DST`
     cp $INITD_SRC $INSTALL_PREFIX$INITD_DST
   fi
+  
+  if [[ $DISTRO = "xenial" ]]; then
+    #Change config path in service unit
+    sed -i 's/sysconfig/default/g' $INSTALL_PREFIX$SYSTEMD_SERVICE_DST
+  fi
 
   log "creating package"
   IFS=',' read -a deps <<< "$PACKAGE_DEPENDENCIES"
@@ -153,7 +164,7 @@ function main() {
   fi
 
   POSTINST_CMD=""
-  if [ $OSQUERY_POSTINSTALL != "" ] && [ -f $OSQUERY_POSTINSTALL ]; then
+  if [[ $OSQUERY_POSTINSTALL != "" ]] && [[ -f $OSQUERY_POSTINSTALL ]]; then
     POSTINST_CMD="--after-install $OSQUERY_POSTINSTALL"
   fi
 ```

Comment (fuzzball5000) [inline tools/deployment/make_linux_package.sh]: Found it, removed it!

Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: Not exactly ;) There's still a single space.

File: tools/deployment/make_linux_package.sh [hunk 8]
```diff
@@ -124,7 +125,12 @@ function main() {
     log "config setup"
     cp $OSQUERY_CONFIG_SRC $INSTALL_PREFIX/$OSQUERY_ETC_DIR/osquery.conf
   fi
-
+
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: Spaces here too :(

File: tools/deployment/make_linux_package.sh [hunk 9]
```diff
@@ -124,7 +125,12 @@ function main() {
     log "config setup"
     cp $OSQUERY_CONFIG_SRC $INSTALL_PREFIX/$OSQUERY_ETC_DIR/osquery.conf
   fi
-
+  
+  if [[ $DISTRO = "xenial" ]]; then
+    #Change config path to Ubuntu/Xenial default
+    SYSTEMD_SYSCONFIG_DST=$SYSTEMD_SYSCONFIG_DST_DEBIAN
+  fi 
+
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: Lots here too

File: tools/deployment/make_linux_package.sh [hunk 10]
```diff
@@ -135,6 +141,11 @@ function main() {
     mkdir -p `dirname $INSTALL_PREFIX$INITD_DST`
     cp $INITD_SRC $INSTALL_PREFIX$INITD_DST
   fi
+
```
Comment (theopolis) [inline tools/deployment/make_linux_package.sh]: Last set of spaces!

**Comment by fuzzball5000** [comment]
image

Forgive me. I'm either blind or insane, or both... ;) But I cannot see those spaces in Atom, and the indentation is consistent with the rest of the file? Two spaces outer, four spaces inner?

**Comment by theopolis** [comment]
I can't speak to what's on your machine, but your master has them: https://github.com/fuzzball5000/osquery/blob/master/tools/deployment/make_linux_package.sh#L144

Perhaps you have not pushed your local changes?

**Comment by fuzzball5000** [comment]
Ok, pulled a fresh copy into a new local repo, and there they are! And as far as I can tell, that's the last of them!

**Comment by theopolis** [comment]
Woot!
